### PR TITLE
Propagate response variables for missing items

### PR DESCRIPTION
**Pull request by poyuki**
not sure is it right extension for this script but probably it will be never merged so i left it there

script for creating default response variable for delivery execution to unblock scoring and provide items to on result page and xml

### CODE REVIEW

File: scripts/tools/PropagateEmptyResponseVariables.php [hunk 1]
```diff
@@ -166,47 +180,64 @@ private function extractAssocAssessmentItemHrefByItemId(AssessmentTest $assessme
...
-            ->setIdentifier('SCORE')
-            ->setCardinality('single')
-            ->setBaseType('float');
 
-        return [$numAttempts, $duration, $response, $completionsStatus, $score,];
+        return [$numAttempts, $duration, $response, $completionsStatus];
     }
 
-    protected function getServiceProxy()
+    protected function getServiceProxy(): ServiceProxy
```
Comment (gabrielfs7) [inline scripts/tools/PropagateEmptyResponseVariables.php:240]: I am getting the following error on local NSA installation

```
root@1defad333fc2:/var/www/html# php index.php 'oat\taoQtiTest\scripts\tools\PropagateEmptyResponseVariables' -de https://nec-deploy.docker.localhost/ontologies/tao.rdf#i6295e701dd95014c461ad8ec95782e1 -wr true

Fatal error: Uncaught TypeError: Return value of oat\taoQtiTest\scripts\tools\PropagateEmptyResponseVariables::getServiceProxy() must be an instance of oat\taoDelivery\model\execution\ServiceProxy, instance of oat\taoDelivery\model\execution\OntologyService returned in /var/www/html/taoQtiTest/scripts/tools/PropagateEmptyResponseVariables.php on line 242

TypeError: Return value of oat\taoQtiTest\scripts\tools\PropagateEmptyResponseVariables::getServiceProxy() must be an instance of oat\taoDelivery\model\execution\ServiceProxy, instance of oat\taoDelivery\model\execution\OntologyService returned in /var/www/html/taoQtiTest/scripts/tools/PropagateEmptyResponseVariables.php on line 242

Call Stack:
    0.0003     387432   1. {main}() /var/www/html/index.php:0
    0.0649     643840   2. oat\tao\model\mvc\Bootstrap->dispatch() /var/www/html/index.php:31
    0.0649     643840   3. oat\tao\model\mvc\Bootstrap->dispatchCli() /var/www/html/tao/models/classes/mvc/Bootstrap.php:246
    0.0665     645160   4. oat\tao\model\routing\CliController->runAction($actionIdentifier = 'oat\\taoQtiTest\\scripts\\tools\\PropagateEmptyResponseVariables', $params = [0 => '-de', 1 => 'https://nec-deploy.docker.localhost/ontologies/tao.rdf#i6295e701dd95014c461ad8ec95782e1', 2 => '-wr', 3 => 'true']) /var/www/html/tao/models/classes/mvc/Bootstrap.php:230
    0.1091     758544   5. call_user_func:{/var/www/html/tao/models/classes/routing/CliController.php:59}($function_name = class oat\taoQtiTest\scripts\tools\PropagateEmptyResponseVariables { private ${oat\oatbox\extension\script\ScriptAction}options = class oat\oatbox\extension\script\OptionContainer { protected $options = [...]; protected $data = [...] }; private ${oat\oatbox\extension\script\ScriptAction}optionsDescription = ['deliveryExecutionId' => [...], 'wetRun' => [...]]; protected $serviceLocator = class oat\oatbox\service\ServiceManager { private $services = [...]; private $configService = class common_persistence_KeyValuePersistence { ... } }; private ${oat\oatbox\extension\AbstractAction}logger = class oat\oatbox\log\LoggerService { protected $logger = NULL; protected $header = NULL; private ${oat\oatbox\service\ConfigurableService}subServices = [...]; private ${oat\oatbox\Configurable}options = [...]; protected $serviceLocator = class oat\oatbox\service\ServiceManager { ... }; private ${oat\oatbox\Configurable}logger = NULL } }, $parameters = [0 => '-de', 1 => 'https://nec-deploy.docker.localhost/ontologies/tao.rdf#i6295e701dd95014c461ad8ec95782e1', 2 => '-wr', 3 => 'true']) /var/www/html/tao/models/classes/routing/CliController.php:59
    0.1091     758624   6. oat\oatbox\extension\script\ScriptAction->__invoke($params = [0 => '-de', 1 => 'https://nec-deploy.docker.localhost/ontologies/tao.rdf#i6295e701dd95014c461ad8ec95782e1', 2 => '-wr', 3 => 'true']) /var/www/html/tao/models/classes/routing/CliController.php:59
    0.1115     826528   7. oat\taoQtiTest\scripts\tools\PropagateEmptyResponseVariables->run() /var/www/html/generis/common/oatbox/extension/script/ScriptAction.php:82
    0.1351     927056   8. oat\taoQtiTest\scripts\tools\PropagateEmptyResponseVariables->getServiceProxy() /var/www/html/taoQtiTest/scripts/tools/PropagateEmptyResponseVariables.php:95


```

Comment (poyuki) [reply scripts/tools/PropagateEmptyResponseVariables.php]: fixed wrong return type, now rely on interface

File: scripts/tools/PropagateEmptyResponseVariables.php [hunk 2]
```diff
@@ -94,22 +96,42 @@ protected function run()
         $variables = $this->createVariables();
         foreach ($deliveryExecutionIdList as $deliveryExecutionId) {
             $deliveryExecution = $this->getServiceProxy()->getDeliveryExecution($deliveryExecutionId);
-            if (in_array($deliveryExecution, self::DE_ENDED_STATUSES, true)) {
-                $outcome[] = sprintf('[%s] Delivery execution not finished', $deliveryExecutionId);
+            if (!in_array($deliveryExecution->getState()->getUri(), self::DE_ENDED_STATUSES, true)) {
```
Comment (gabrielfs7) [inline scripts/tools/PropagateEmptyResponseVariables.php:99]: This means this if always passed yesterday? 

```
if (in_array($deliveryExecution, self::DE_ENDED_STATUSES, true)) {
```

Comment (poyuki) [reply scripts/tools/PropagateEmptyResponseVariables.php:99]: but for our case with deliveries if they have `test variables` in result they should be finished, so we good and i check those 8 which not passed they are terminated
so we didn't affect anything unexpected
but yes check was broken :(

Comment (gabrielfs7) [reply scripts/tools/PropagateEmptyResponseVariables.php:99]: Yes, we only executed to isolated Delivery Executions that did not have any response before.

**Pull request closed without merging**